### ISSUE TITLE
[discussion] Repology support added, possible CONTROL format improvements

### ISSUE THREAD

**Issue by AMDmi3** [microsoft/vcpkg#2011]
Hi!

Just wanted to drop a note that vcpkg is now supported by [Repology](https://repology.org) (roughly, it's a service which compares package versions in A LOT of repositories and discovers outdated packages). The list of [outdated packages](https://repology.org/metapackages/outdated-in-repo/vcpkg/) would probably be most useful to vcpkg contributors.

Repology parses package versions and descriptions from CONTROL files, and I'd like to suggest some improvements to the format of this file which would make life easier for repology and probably vcpkg users/developers as well.

- Split revision into separate field, e.g.
  ```
  - Version: 2.0-1
  + Version: 2.0
  + Revision: 1
  ```
   this way upstream version would be more clearly distinguishable from port revision (repology tries to remove revision by dropping everything after `-`, but this breaks on e.g. `2017-01-01` versions). This also make vcpkg more future proof, e.g. adding `Epoch` field would be easier if it's needed at some moment.

- Add `Homepage` field. It's often useful to get to project's homepage for docs, filing an issue or checking for a new version, and now only place it is available it is a `portfile.cmake`, from where it's hard to parse for both machines and humans. Repology can make use of the homepage as well, matching differently named packages which are actually the same thing, and using a homepage to split different packages with the same name. It can also report broken homepage links, and some time it'll learn to look for new versions at upstream homepages.

**Comment by cbezault** [comment]
We've added homepages to the control files.

**Comment by AMDmi3** [comment]
> We've added homepages to the control files.

These are now available on repology.org

**Comment by JonLiu1993** [comment]
Hi @AMDmi3 
Thanks for posting this issue. In my opinion. the current status might be enough.
So I'm closing this issue for now.
